File: frontend/src/api/health.ts
import type { HealthResponse } from '../bindings/HealthResponse'
import { isRecord, request } from './client'
export function parseHealth(data: unknown): HealthResponse {
  if (!isRecord(data) || data.status !== 'ok' || typeof data.service !== 'string' || typeof data.version !== 'string') {
    throw new Error('Invalid health response')
  }
  return { status: data.status, service: data.service, version: data.version }
}
export function getHealth(): Promise<HealthResponse> { return request('/api/health', parseHealth) }

File: src/controllers/health.rs
use loco_rs::prelude::*;
use mobile_qa_contracts::browser::{HealthResponse, HealthStatus};
#[debug_handler]
async fn current() -> Result<Response> {
    format::json(HealthResponse {
        status: HealthStatus::Ok,
        service: "mobile-qa".into(),
        version: env!("CARGO_PKG_VERSION").into(),
    })
}
async fn unknown() -> axum::http::StatusCode {
    axum::http::StatusCode::NOT_FOUND
}
pub fn routes() -> Routes {
    Routes::new()
        .prefix("/api")
        .add("/health", get(current))
        .add("/", axum::routing::any(unknown))
        .add("/{*path}", axum::routing::any(unknown))
}
